reference/features/security.md

```diff
@@ -0,0 +1,272 @@
+      ],
+      "indexes": [
+          "*"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    },
+    {
+      "description": "Default Admin API Key (Use it for all other operations. Caution! Do not share it on the client side)",
+      "key": "380689dd379232519a54d15935750cc7625620a2ea2fc06907cb40ba5b421b6f",
+      "actions": [
+          "*"
+      ],
+      "indexes": [
+          "*"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    }
+  ]
+}
+```
+
+### Updating an API key
+
+You can freely update an API key until it expires. This includes editing the indexes, endpoints, and routes it can access, as well as its description and expiry date.
+
+To update an API key, you must use the [`PATCH /keys/:key`](/reference/api/keys.md#update-a-key) endpoint which can only be accessed with the `master` key. Here, `:key` stands for the full value of the key you want to update.
+
+#### Example
+
+After creating your `Manage sci-fi books key` with an expiry date set for the next day, you decide to extend its duration in case you need to update some of the books:
+
+<CodeSamples id="security_guide_update_key_1" />
+
+```json
+{
+  "description": "Manage sci-fi books key",
+  "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+  "actions": [
+    "documents.add",
+    "documents.get",
+    "documents.delete"
+  ],
+  "indexes": [
+  "scifi_books"
+  ],
+  "expiresAt": "2022-03-01T00:00:00Z",
+  "createdAt": "2022-01-01T10:00:00Z",
+  "updatedAt": "2022-01-01T10:00:00Z"
+}
+```
+
+As you can see, MeiliSearch supports partial updates with the `PATCH` route. This means that your payload can contain only the data you want to update—in this case, `expiresAt`.
+
+### Deleting an API key
+
+If a key is no longer useful or has been compromised, you can use [`DELETE /keys/:key`](/reference/api/keys.md#delete-a-key) to disable it before its expiry date.
+
+#### Example
+
+Since you don't expect to routinely update your instance's indexes after adding the first batch of documents, you decide the default admin key MeiliSearch is not necessary:
+
+<CodeSamples id="security_guide_delete_key_1" />
+
+### Expired keys
+
+Once a key is past its `expiresAt` date, it becomes inaccessible and does not show up when querying [`GET /keys` or `GET /keys/:key`](/reference/api/keys.md). It is not currently possible to renew a key after its expiry date.
+
+## Communicating with a protected instance
+
+After an instance is secured, only the [`GET /health` endpoint](/reference/api/health.md) will be publicly available. To access any other API endpoint, you must add an API key with suitable permissions to the request header.
+
+We strongly discourage using the `master` key for API calls. Doing so risks publicly exposing your `master` key and giving malicious users full access over your MeiliSearch indexes.
```

Comment:
Would be nice to provide a code example showing what NOT to do.

Also, there is one big exception to this, which is the API key management endpoints right?

Reply:
Good question. Maybe something like this 

❌  BAD

```js
// WARNING:  if the maste key is leaked, you have to restart MeiliSearch with a new master key
// All your other API keys are automatically obsolete.
const masterClient = new Meilisearch({ host: "http://localhost:7700", apiKey: "masterKey")
```

✅  Good
```js
// We use the private key listed in `getKeys` or a generated key. 
// In case of leak, it can be changed using `updateKey`
const client = new Meilisearch({ host: host: "http://localhost:7700", apiKey: secretKey })
```